### ISSUE TITLE
cargo install --git https://github.com/gfx-rs/naga naga-cli fails

### ISSUE THREAD

**Issue by TrueDoctor** [gfx-rs/naga#2385]
Trying to execute
```
cargo install --git https://github.com/gfx-rs/naga naga-cli
```
fails with 
```
Updating git repository `https://github.com/gfx-rs/naga`
error: package `/home/dennis/.cargo/git/checkouts/naga-dbb2b19faed49210/f31093f/xtask/hlsl-snapshots/Cargo.toml` is a member of the wrong workspace
expected: /home/dennis/.cargo/git/checkouts/naga-dbb2b19faed49210/f31093f/Cargo.toml
actual:   /home/dennis/.cargo/git/checkouts/naga-dbb2b19faed49210/f31093f/xtask/Cargo.toml
```

Environment:
```
rustc 1.70.0 (90c541806 2023-05-31)

cargo 1.70.0 (ec8a8a0ca 2023-04-25)
release: 1.70.0
commit-hash: ec8a8a0cabb0e0cadef58902470f6c7ee7868bdc
commit-date: 2023-04-25
host: x86_64-unknown-linux-gnu
libgit2: 1.6.3 (sys:0.17.0 vendored)
libcurl: 8.0.1-DEV (sys:0.4.61+curl-8.0.1 vendored ssl:OpenSSL/1.1.1t)
os: Arch Linux Rolling Release [64-bit]
```

**Comment by teoxoy** [comment]
@ErichDonGubler any idea why?

**Comment by fornwall** [comment]
Should be fixed by #2398.

**Comment by ErichDonGubler** [comment]
@teoxoy: Sorry, I've been pretty backed up in my work queue for the last few weeks! I think @fornwall's already diagnosed the issue; it appears that Cargo does not take kindly to having a crate be marked as members of multiple workspaces. 

Thanks for finding and fixing, @fornwall!
